OKR.md

```diff
@@ -0,0 +1,52 @@
+# Quarterly Objectives and Key Results
+
+We try to frame our ongoing work using a process based on quarterly Objectives and Key Results (OKRs). Objectives reflect outcomes that are challenging, but realistic. Results are tangible and measurable.
+
+## 2018 Q4
+
+**go-ipfs handles large datasets (1TB++) without a sweat**
+- `PX` - It takes less than 48 hours to transfer 1TB dataset over Fast Ethernet (100Mbps)
+- `PX` - It takes less than 12 hours to transfer 200GB sharded dataset over Fast Ethernet (100Mbps)
+- `PX` - There is a prototype implementation of GraphSync for go-ipfs
+- `PX` - There is a better and more performant datastore module (e.g Badger or better)
+- `P1` - Rewrite pinning data structures to support large data sets / many files performantly
+
+**The bandwidth usage is reduced significantly and is well kept under control**
+- `PX` - Users can opt out of providing every IPLD node (and only provide root hashes)
+- `PX` - "Bitswap improvements around reducing chattiness, decreasing bandwidth usage (fewer dupe blocks), and increasing throughput"
+
+**It is a joy to use go-ipfs programatically**
+- `PX` @magik6k - The Core API is finalized and released. Make it easier to import go-ipfs as a package
+- `PX` - go-ipfs-api exposes the new Core API
+- `PX` - go-ipfs Daemon uses the new Core API
+- `PX` - go-ipfs Gateway uses the new Core API
+- `PX` - go-ipfs-cms uses uses the new Core API
+- `PX` - The legacy non Core API is deprecated and the diagram on go-ipfs README is updated
+
+**go-ipfs becomes a well maintained project**
+- `PX` - The Go Contributing Guidelines are updated to contemplate expecations from Core Devs and instructions on how to be an effective contributor (e.g include PR templates)
+- `PX` - A Lead Maintainer Protocol equivalent is proposed, reviewed by the team, merged and implemented
+- `PX` - Every issue on https://waffle.io/ipfs/go-ipfs gets triaged (reviewed and labeled following https://github.com/ipfs/pm/blob/master/GOLANG_CORE_DEV_MGMT.md)
+- `PX` - Every package has tests and tests+code coverage are running on Jenkins
+- `PX` - There is an up-to-date Architecture Diagram of the Go implementation of IPFS that links packages to subsystems to workflows
+
+**gx becomes a beloved tool by the Go Core Contributors**
+- `PX` - 
+- `PX` - 
```

Comment:
@Stebalien I know there are a lot of new goals for GX but I really can't tell what these are. Could you describe the ones planned for Q4 as KRs?

Reply:
@warpfork mentioned this in his retrospective comment - do you know if there's a tracking issue?

Reply:
The thing I remember hearing is semver based heuristics for flattening dependencies, so you'll get a minimal set of imports of any given package. In his retro @warpfork mentioned changes that keep gx from doing so much import rewriting - I'm not sure how that would work but that sounds neat.

Reply:
Steven suggested these KRs for gx:
- You can update a minor version of a transitive dependancy without updating intermediate dependancies
- Imports are not rewritten in go-ipfs

Reply:
@whyrusleeping who had thoughts on this

Reply:
ping to both @whyrusleeping and @Stebalien to iterate on gx KRs that would measure success of this tool being productive/efficient

Reply:
KR: update a transitive dependency without touching intermediate dependencies
KR: go-ipfs doesn't have checked-in gx paths